### PR TITLE
feat: also synchronize additional dependencies for Python hooks

### PR DESCRIPTION
**Pull request by noirbizarre**
Hey 👋🏼 

Here's the additional dependencies support.

I have been in another direction from [your branch](https://github.com/GabDug/sync-pre-commit-lock/tree/experiment/additional_dependencies), but I'll update it with ~the debug statements and some of the missing error handling~ (done).

Sorry it was a big one, but I start I tried as much as possible not to change the signatures, but at some point it was just adding more complexity (eg., `mapping`, `mapping_reverse_by_url` as params,...)

### PDM
image

### Poetry
image

> [!NOTE]
> The structure/algorithm make it easy to also handle repository renaming (like `ruff`).
> I'll submit it in another pull requests later as it is already done too but based on this PR

> [!NOTE]
> I added 2 dev scripts, `pdm test` and `pdm test-all` that were actually useful to me so I decided too submit them in the process, but I can totally remove them

> [!NOTE]
> Not done in this PR (as it is already a big one) but given `packging` is now a dependency, maybe it's worth replacing the [`GenericLockedPackage` type](https://github.com/GabDug/sync-pre-commit-lock/blob/main/src/sync_pre_commit_lock/actions/sync_hooks.py#L15-L18) by [`packging.requirements.Requirement`](https://packaging.pypa.io/en/latest/requirements.html)

### CODE REVIEW

**Review by GabDug — CHANGES_REQUESTED** [review]
Hey @noirbizarre!

Thanks for this well-crafted PR! No worries about changing the signature; it’s not a big deal.

I just have one suggestion: could you please update the README to include the new feature and the dependency on `packaging`? This PR should also check off one item on the TODO list!

Aside from that, it looks good to go. I'll likely wait for your other PR to be merged before releasing version 0.7.0.

Thanks again!

**Comment by noirbizarre** [comment]
README updated 👍🏼 
I also added a checkbox for the repository URL update support.

Agreed for the release. It's ready, I'll submit the PR ASAP after this one is merged 👍🏼

Thanks to you, great tool, very useful !

**Review by GabDug — APPROVED** [review]

**Pull request merged**
